pkg/volume/csi/csi_plugin.go

```diff
@@ -156,9 +161,118 @@ func (h *RegistrationHandler) RegisterPlugin(pluginName string, endpoint string,
 		return err
 	}
 
+	if utilfeature.DefaultFeatureGate.Enabled(features.MutableCSINodeAllocatableCount) {
+		stopCh := make(chan struct{})
+		h.allocatableUpdateChannels.Store(pluginName, stopCh)
+		go startPeriodicAllocatableUpdate(pluginName, stopCh)
```

Comment:
Can a rogue plugin or some error condition cause `RegisterPlugin` to get called multiple times for same plugin? 
I am slightly wary of goroutines that are basically not tracked via some kind of pool.

Reply:
If we are really going to start multiple goroutines and keep track of them, it might be much better idea to use something like - https://github.com/kubernetes/kubernetes/blob/master/pkg/util/goroutinemap/goroutinemap.go

Reply:
That makes sense, latest rev implements a [Swap](https://cs.opensource.google/go/go/+/refs/tags/go1.24.1:src/sync/map.go;l=358) and close pattern to ensure there is only one goroutine running per plugin at a time.

Reply:
Thank you. I personally think, we need an "owner" for this set of goroutines (lets call it, `CSINodeUpdater`). And by owner I mean, the way - kubelet owns volumanagaer as a property, volumemanager owns desiredstateofworldpopulator as a property, so on and so forth.

Having package global variable (unexported) seems iffy and is harder to reason about and keep track of.

Reply:
x-ref: https://github.com/kubernetes/kubernetes/pull/130007/commits/ebaf877b12e9805077b154a087639087b1b60938